crates/chain-config/src/config/state/writer.rs

```diff
@@ -308,72 +277,105 @@ impl StateWriter {
         da_height: DaBlockHeight,
     ) -> anyhow::Result<()> {
         match &mut self.encoder {
-            EncoderType::Json { buffer, .. } => {
-                buffer.block_height = height;
-                buffer.da_block_height = da_height;
+            EncoderType::Json { builder, .. } => {
+                builder.set_block_height(height);
+                builder.set_da_block_height(da_height);
                 Ok(())
             }
             #[cfg(feature = "parquet")]
             EncoderType::Parquet {
                 block_height,
                 da_block_height,
+                compression,
                 ..
             } => {
-                block_height.write(vec![height])?;
-                da_block_height.write(vec![da_height])?;
+                Self::write_single_el_parquet(block_height, height, *compression)?;
+                Self::write_single_el_parquet(da_block_height, da_height, *compression)?;
+
                 Ok(())
             }
         }
     }
 
-    pub fn close(self) -> anyhow::Result<()> {
+    #[cfg(feature = "parquet")]
+    fn write_single_el_parquet(
+        path: &std::path::Path,
+        data: impl serde::Serialize,
+        compression: ZstdCompressionLevel,
+    ) -> anyhow::Result<()> {
+        let mut encoder = parquet::encode::Encoder::new(
+            std::fs::File::create(path)?,
+            compression.into(),
+        )?;
+        encoder.write(vec![postcard::to_stdvec(&data)?])?;
+        encoder.close()
+    }
+
+    pub fn close(self) -> anyhow::Result<SnapshotMetadata> {
         match self.encoder {
-            EncoderType::Json {
-                buffer,
-                state_file_path,
-            } => {
-                let file = std::fs::File::create(state_file_path)?;
-                serde_json::to_writer_pretty(file, &buffer)?;
-                Ok(())
+            EncoderType::Json { builder } => {
+                let state_config = builder.build()?;
+
+                let state_file_path = self.dir.join("state_config.json");
```

Comment:
Should we use a public constant? With the `metadata.json` no expectations are placed on the naming of the state config file. Whatever is the value of the `state_config` json-object key is going to be used as the path from which to read the `StateConfig`. 

As such maybe it would be better to keep the name we're currently using (`state_config.json`) an implementation detail housed solely in `close` where it's used? As such maybe to delete the already existing constant `STATE_CONFIG_FILENAME`?